src/structures/Guild.js

```diff
@@ -468,22 +468,38 @@ class Guild {
     return this.client.resolver.resolveGuildMember(this, user);
   }
 
+  /**
+   * An object containing information about a guild member's ban.
+   * @typedef {Object} BanInfo
+   * @property {User} user User that was banned
+   * @property {?string} reason Reason the user was banned
+   */
+
+  /**
+   * Fetch a ban for a user.
+   * @returns {Promise<BanInfo>}
+   * @param {UserResolvable} user The user to fetch the ban for
+   * @example
+   * // Get ban
+   * guild.fetchBan(message.author)
+   *   .then(({ user, reason }) => console.log(`${user.tag} was banned for the reason: ${reason}.`))
+   *   .catch(console.error);
+   */
+  fetchBan(user) {
+    return this.client.rest.methods.getGuildBan(this, user);
+  }
+
   /**
    * Fetch a collection of banned users in this guild.
-   * @returns {Promise<Collection<Snowflake, User>>}
+   * @returns {Promise<Collection<Snowflake, BanInfo>>}
```

Comment:
Changing the return value like that is sadly breaking. (semver: major)

Maybe adding a parameter like `withReasons` defaulting to `false` could be used to work around that?

Reply:
Ah okay, fair enough and yeah that sounds good.